global.go

```diff
@@ -113,58 +98,64 @@ func NewStdLogAt(l *Logger, level zapcore.Level) (*log.Logger, error) {
 // It returns a function to restore the original prefix and flags and reset the
 // standard library's output to os.Stdout.
 func RedirectStdLog(l *Logger) func() {
-	flags := log.Flags()
-	prefix := log.Prefix()
-	log.SetFlags(0)
-	log.SetPrefix("")
-	logger := l.WithOptions(AddCallerSkip(_stdLogDefaultDepth + _loggerWriterDepth))
-	logFunc := logger.Info
-	log.SetOutput(&loggerWriter{logFunc})
-	return func() {
-		log.SetFlags(flags)
-		log.SetPrefix(prefix)
-		log.SetOutput(os.Stderr)
+	f, err := redirectStdLogAt(l, InfoLevel)
+	if err != nil {
+		// Can't get here, since passing InfoLevel to redirectStdLogAt always
+		// works.
+		panic(err)
 	}
+	return f
 }
 
 // RedirectStdLogAt redirects output from the standard library's package-global
-// logger to the supplied logger at required level. Since zap already handles
-// caller annotations, timestamps, etc., it automatically disables the standard
-// library's annotations and prefixing.
+// logger to the supplied logger at the specified level. Since zap already
+// handles caller annotations, timestamps, etc., it automatically disables the
+// standard library's annotations and prefixing.
 //
 // It returns a function to restore the original prefix and flags and reset the
 // standard library's output to os.Stdout.
 func RedirectStdLogAt(l *Logger, level zapcore.Level) (func(), error) {
+	return redirectStdLogAt(l, level)
+}
+
+func redirectStdLogAt(l *Logger, level zapcore.Level) (func(), error) {
 	flags := log.Flags()
 	prefix := log.Prefix()
 	log.SetFlags(0)
 	log.SetPrefix("")
 	logger := l.WithOptions(AddCallerSkip(_stdLogDefaultDepth + _loggerWriterDepth))
-	var logFunc func(string, ...zapcore.Field)
-	switch level {
+	logFunc, err := levelToFunc(logger, level)
+	if err != nil {
+		return nil, err
+	}
+	log.SetOutput(&loggerWriter{logFunc})
+	return func() {
+		log.SetFlags(flags)
+		log.SetPrefix(prefix)
+		log.SetOutput(os.Stderr)
+	}, nil
+}
+
+func levelToFunc(logger *Logger, lvl zapcore.Level) (f func(string, ...zapcore.Field), err error) {
```

Comment:
This felt like a place where named return values simplified the code; happy to remove them if any reviewers disagree.

Reply:
not strong objections, but I don't think it provides a huge amount of value over
```
return logger.Debug, nil
```

it's slightly less verbose, but doesn't really affect clarity otherwise, so i'd prefer to avoid.

Reply:
Fair enough, I'm not strongly attached to the named results.